ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/ChangeId.scala

```diff
@@ -3,14 +3,45 @@
 
 package com.daml.ledger.participant.state.v2
 
+import com.daml.lf.crypto.Hash
 import com.daml.lf.data.Ref
 
 /** Identifier for ledger changes used by command deduplication
   *
   * @see ReadService.stateUpdates for the command deduplication guarantee
   */
-final case class ChangeId(
-    private val applicationId: Ref.ApplicationId,
-    private val commandId: Ref.CommandId,
-    private val actAs: Set[Ref.Party],
-)
+final class ChangeId(
```

Comment:
Because equality is based on the hash, not on the actual arguments to the constructor. It's similar to `GlobalKey`.

Reply:
This class is similar to `GlobalKey`, which isn't a case class either. When I see a `case class`, I expect equality to be defined in terms of the constructor arguments.

Reply:
I think the situation is a bit different. For GlobalKey the constructor is private and the hash is an argument to the constructor. Here you derive the hash in the constructor and the constructor is public.

Reply:
Can you elaborate why this difference matters here? `GlobalKey` has a public `apply` method which is as good as a constructor, isn't it?

Reply:
> provided you don’t assume your hash is broken.

That's precisely why I define equality using the hash, namely so that hash collisions are accounted for. But I won't fight over this. I'll turn it into a case class.

Reply:
Fine to keep it as well, I mostly wanted to understand the motivation.